src/main/java/View/playlistView.java

```diff
@@ -22,19 +22,31 @@ public class playlistView extends JFrame implements ActionListener {
     private JButton searchButton;
     private JButton newPlaylistButton;
     private JLabel title;
-    private Font font;
     private boolean spacePlaying;
     private JTextField searchBar;
     private musicEngineControllerViewModel musicEngineControllerViewModel;
     private searchViewModel searchViewModel;
     private PlaylistPanelView playlistPanel;
     private PlayBar playBar;
 
+    private ImageIcon icon;
+    private ImageIcon logoImg;
+    private ImageIcon logoSmall;
+    private JLabel logo;
+
+    private int DEFAULT_WIDTH = 130;
+    private int DEFAULT_HEIGHT = 40;
+    private int DEFAULT_KERNING = 20;
+
+
     /**
      * constructor
      * @param user takes in the user's data to display their own dashboard!
      */
-    public playlistView(InMemoryUser user){
+    public playlistView(InMemoryUser user, ImageIcon icon, ImageIcon logoImg, ImageIcon logoSmall){
```

Comment:
The null pointer exception thrown due to the home button in search is due to this change here - search output view uses the second overloaded constructor which does not have these icon parameters - then presumably you try to access the icon objects but they don't exist in playlistView objects made by searchOutputView.

Reply:
Hm, okay. I'll fix that later then. I'm thinking of changing it slightly anyway